### PR TITLE
feat(BLM): use 'main' instead of 'master'

### PR DESCRIPTION
**Pull request by maia-tae**
Add support for publishing with **main** branch to follow on the trend to remove slavery references from tech jargon.

image

The default release will continue to detect 'master' branch so this is not a breaking change.

### CODE REVIEW

**Review by henri-hulski — COMMENTED** [review]
Great idea!
Are you sure it's not breaking? 
Have you left some tests to check if it works also with master?

File: src/sequences/defaultRelease.js [hunk 1]
```diff
@@ -45,14 +55,8 @@ export const defaultReleaseSequence = [
...
   {
-    master: [
-      cook.tagCurrentCommit,
-      cook.pushTagToRemote,
-      cook.byReleaseTarget,
-      {
-        github: cook.createGithubRelease,
-      },
-    ],
+    master: publishToGithub,
```
Comment (maia-tae) [inline src/sequences/defaultRelease.js:58]: @henri-hulski This is the only place where we need "master" for the rest we use tags to detect the last release.

File: src/sequences/defaultRelease.js [hunk 2]
```diff
@@ -37,6 +46,7 @@ export const defaultReleaseSequence = [
   cook.byBranch,
   {
     master: cook.mapTemporaryNpmTagTo('latest'),
+    main: cook.mapTemporaryNpmTagTo('latest'),
```
Comment (maia-tae) [inline src/sequences/defaultRelease.js:49]: And here...

File: src/sequences/defaultRelease.js [hunk 3]
```diff
@@ -45,14 +55,8 @@ export const defaultReleaseSequence = [
...
-    master: [
-      cook.tagCurrentCommit,
-      cook.pushTagToRemote,
-      cook.byReleaseTarget,
-      {
-        github: cook.createGithubRelease,
-      },
-    ],
+    master: publishToGithub,
+    main: publishToGithub,
```
Comment (maia-tae) [inline src/sequences/defaultRelease.js:59]: ... and this other reference here.

@henri-hulski This is all we need. For the rest, we detect current release with tags (which has always seemed safer).

**Comment by maia-tae** [comment]
Writing a test to ensure publication to github works for 'main' and 'master'.

**Comment by maia-tae** [comment]
I tried to write tests for publication to 'main' and 'master' and it is taking too long (we do not have any test for the details of the default release yet).

**Comment by maia-tae** [comment]
Sorry... I made a mess while pushing the changes.

I updated the commit to remove unused arguments and fix tests so that we see more clearly  where "master" and "main" are referenced.

Should be all good now. Effectively, the only uses are in the default release branching and in the README for linking to some code.

**Review by henri-hulski — COMMENTED** [review]
Thanks for explaining and clean up 💗
Makes sense so let's give it a try. 

It seems you have added here also the simplify release note commit which is already released. 
I think you need to rebase somehow to get rid of it. 😜

**Comment by maia-tae** [comment]
This should merge into 'main'. Closing.

**Pull request closed without merging**